### PR TITLE
Added sceditor typings as a first start

### PR DESCRIPTION
**Pull request by mnp-mid**
Added sceditor typings as a first start. I intend to add more typings in future.

### CODE REVIEW

File: types/sceditor/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,106 @@
+// Type definitions for SCEditor 2.1.2
```
Comment (andy-ms) [inline types/sceditor/index.d.ts]: Just `2.1`, not `2.1.2`

File: types/sceditor/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+// Project: https://www.sceditor.com/
+// Definitions by: Martin Pfund
```
Comment (andy-ms) [inline types/sceditor/index.d.ts]: `// Definitions by: Martin Pfund <https://github.com/mnp-mid>`

File: types/sceditor/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+    interface ISCEditorOptions {
+        plugins?: string;
+        format?: string;
+        toolbar?: string;
+        locale?: string;
+    }
+
+    interface ISCEditorInstance {
+        // _(...args: string[]): string;
```
Comment (andy-ms) [inline types/sceditor/index.d.ts]: Are are all of these commented out?

File: types/sceditor/index.d.ts [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+// Project: https://www.sceditor.com/
+// Definitions by: Martin Pfund
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.6
+
+// Work In Progress, may contains errors
+
+// declare module sceditor {
+declare module app.common.sceditor {
+    interface ISCEditor {
```
Comment (andy-ms) [inline types/sceditor/index.d.ts]: Avoid beginning type names with `I` unnecessarily.

File: types/sceditor/tslint.json [hunk 5]
```diff
@@ -0,0 +1,79 @@
+{
+    "extends": "dtslint/dt.json",
+    "rules": {
```
Comment (andy-ms) [inline types/sceditor/tslint.json]: Please remove this and make this file just `{ "extends": "dtslint/dt.json" }` -- new packages shouldn't need lint rule disables.

**Review by ghost — CHANGES_REQUESTED** [review]

File: types/sceditor/index.d.ts (hunk 1 above)
Comment (mnp-mid) [reply types/sceditor/index.d.ts]: ok

File: types/sceditor/index.d.ts (hunk 4 above)
Comment (mnp-mid) [reply types/sceditor/index.d.ts]: ok

**Comment by mnp-mid** [comment]
Hi andy-ms,

I have updated files as requested. The reason why most of the content is commented out is, that I didn't want to put quite some effort into something that might be rejected at the end for any reason. I am using the JavaScript Project SCEditor locally for my projects and I extend the interface for my own need and so I wanted to share the file with everybody and continue to let it grow slowly. 

What can I do in additon to let type definition pass? This is what I still get.

Error: At 9:24 : Expected /<([^>]+)> */
    at Object.parseHeaderOrFail (/home/travis/build/DefinitelyTyped/DefinitelyTyped/node_modules/definitelytyped-header-parser/index.js:45:15)
    at /home/travis/build/DefinitelyTyped/DefinitelyTyped/node_modules/types-publisher/src/lib/definition-parser.ts:93:3
    at Generator.next (<anonymous>)
    at fulfilled (/home/travis/build/DefinitelyTyped/DefinitelyTyped/node_modules/types-publisher/bin/lib/definition-parser.js:4:58)
    at <anonymous>
Error: Parsing failed.

File: types/sceditor/index.d.ts (hunk 3 above)
Comment (mnp-mid) [reply types/sceditor/index.d.ts]: Yes, this is a first start only. I plan to put more definitions into it.

**Comment by sheetalkamat** [comment]
You need to add test. also I see that in the definition you just have one const.. But the first commit in this showed that var type being object? So is string the right type?

**Review by sheetalkamat — CHANGES_REQUESTED** [review]
You need to add test. also I see that in the definition you just have one const.. But the first commit in this showed that var type being object? So is string the right type?

**Review by mnp-mid — COMMENTED** [review]
Readded files

**Comment by mnp-mid** [comment]
I almost forgot about this. I have readded the files for review. As mentioned already, I would put more definitions into the index file once this goes through, however I always run into trouble with Travis CI builds which I just don't see what to do.

**Comment by mnp-mid** [comment]
Created a new pullrequest with #23948

**Pull request closed without merging**